storage/mysql/admin_storage.go

```diff
@@ -320,6 +331,11 @@ func (t *adminTX) CreateTree(ctx context.Context, tree *trillian.Tree) (*trillia
 		return nil, fmt.Errorf("unexpected DuplicatePolicy value: %v", newTree.DuplicatePolicy)
 	}
 
+	privateKey, err := proto.Marshal(newTree.PrivateKey)
```

Comment:
I've added some validation that attempts to unmarshal the `Any`, which at least checks that there's a proto in there. Proper validation of key will have to come later, once I've added the code that takes this `Any` and returns a `crypto.Signer`. I think that'd have to take the form of attempting to produce an STH. In the case of updating the private key, this STH could be verified using the old key to check that the key itself hasn't changed (only the provider of the key).